File: Lol2019/Day_6/B/3.cpp
#include <cstdio>
#include <algorithm>

using namespace std;

const int BOUND = 1e4;

int calc(pair<int, int> A, pair<int, int> B) {
    return max(abs(A.first - B.first), abs(A.second - B.second));
}

int main() {
    pair<int, int> A, B, zero = {0, 0};
    scanf("%d%d%d%d", &A.first, &A.second, &B.first, &B.second);
    int best_dist = -1;
    for (int x = -BOUND; x <= BOUND; ++x)
        for (int y = -BOUND; y <= BOUND; ++y) {
            int dist = calc(A, {x, y}) + calc(zero, {x, y}) + calc(B, {x, y});
            if (best_dist == -1 || dist < best_dist)
                best_dist = dist;
        }
    printf("%d", best_dist);
}

File: Lol2019/Day_6/B/3.py
a, b = map(int, input().split())
c, d = map(int, input().split())

dist = 10 ** 9
for i in range(-10000, 10001):
    L1 = abs(a - i)
    L2 = abs(c - i)
    L3 = abs(i)
    for j in [b + L1, b - L1, d + L2, d - L2, L3, -L3]:
        dist = min(dist, max(abs(a - i), abs(b - j)) + max(abs(c - i), abs(d - j)) + max(abs(i), abs(j)))

print(dist)
